crates/wash-runtime/src/host/http.rs

```diff
@@ -250,37 +411,41 @@ impl HostPlugin for HttpServer {
         Ok(())
     }
 
-    async fn on_workload_unbind(
-        &self,
-        workload_id: &str,
-        _interfaces: HashSet<WitInterface>,
-    ) -> anyhow::Result<()> {
-        debug!(workload_id, "removing HTTP workload handle");
-
+    async fn on_workload_unbind(&self, workload_id: &str) -> anyhow::Result<()> {
         // Remove from workload handles
         let mut handles_guard = self.workload_handles.write().await;
-        handles_guard.retain(|_, (handle, _, _)| handle.id() != workload_id);
+        handles_guard.remove(workload_id);
+        drop(handles_guard);
 
-        // Remove from workload configs
-        self.workload_configs.write().await.remove(workload_id);
+        self.router.on_workload_unbind(workload_id).await?;
```

Comment:
Workload handles are removed before notifying the router, creating a window where requests could be routed to a missing workload.

```suggestion
        // Remove from workload handles
        self.router.on_workload_unbind(workload_id).await?;
        self.workload_handles.write().await.remove(workload_id);
```

Reply:
good call